tests/test_core/test_alignment.py

```diff
@@ -1933,6 +1933,15 @@ def test_copy(self):
         copied = aln.copy()
         self.assertEqual(aln.info, copied.info)
 
+    def test_deep_copy(self):
+        """correctly deep copy aligned objects in an alignment"""
+        path = "data/brca1_5.paml"
+        aln = load_aligned_seqs(path, array_align=False, moltype="dna")
+        aln = aln[20:30]
+        for seq in aln.seqs:
+            new_seq = seq.deep_copy()
+            self.assertEqual(len(new_seq.data), 10)
+
```

Comment:
add 3 annotations onto a sequence:
1. one that lies within the slice (e.g. 22, 25)
2. one that straddles the slice boundary (e.g. 25, 35)
3. one outside (e.g. 10, 20)

The deepcopy seq should (1) and (2), but not 3. The `get_slice()` should also produce the correct sequence.